velox/common/memory/MemoryArbitrator.cpp

```diff
@@ -497,6 +497,17 @@ const MemoryArbitrationContext* memoryArbitrationContext() {
   return arbitrationCtx;
 }
 
+ScopedMemoryArbitration::ScopedMemoryArbitration(MemoryPool* pool)
+    : pool_(pool), ctx_{pool} {
+  VELOX_CHECK_NOT_NULL(pool_);
+  VELOX_CHECK(pool_->isLeaf());
+  pool_->enterArbitration();
+}
+
+ScopedMemoryArbitration::~ScopedMemoryArbitration() {
+  pool_->leaveArbitration();
```

Comment:
enterArbitration and leaveArbitration are both changed to protected methods now. You cannot call them here?

Reply:
They are annotated as friend methods.